apps/src/codebridge/utils/validateFileName.ts

```diff
@@ -0,0 +1,55 @@
+import {ProjectType, FolderId, ProjectFile} from '@codebridge/types';
+
+import codebridgeI18n from '@cdo/apps/codebridge/locale';
+
+import {isDuplicateFileName, DuplicateFileError} from './isDuplicateFileName';
+import {isValidFileName} from './isValidFileName';
+
+type ValidateFileNameArgs = {
+  fileName: string;
+  folderId: FolderId;
+  projectFiles: ProjectType['files'];
+  isStartMode: boolean;
+  validationFile: ProjectFile | undefined;
+};
+
+export const validateFileName = ({
+  fileName,
+  folderId,
+  projectFiles,
+  isStartMode,
+  validationFile,
+}: ValidateFileNameArgs) => {
+  if (!fileName.length) {
+    return;
+  }
+
+  const [, extension] = fileName.split('.');
+  if (!extension) {
+    return codebridgeI18n.noFileExtensionError();
+  }
+
+  if (!isValidFileName(fileName)) {
+    return codebridgeI18n.invalidNameError();
+  }
+
+  const duplicateFileError = isDuplicateFileName({
+    fileName,
+    folderId,
+    projectFiles,
+    isStartMode,
+    validationFile,
+  });
+
+  if (duplicateFileError) {
+    if (duplicateFileError === DuplicateFileError.DUPLICATE_FILE) {
+      return codebridgeI18n.duplicateFileError({fileName});
+    } else if (
+      duplicateFileError === DuplicateFileError.DUPLICATE_SUPPORT_FILE
+    ) {
+      return codebridgeI18n.duplicateSupportFileError({fileName});
+    } else {
```

Comment:
if we use a switch statement on the enum do we need to throw an error here? Also should we throw an error given that could crash the page? The return value should always be one of the enum values or false right?

Reply:
I changed the logic on it. My original thought was that it _should_ crash the page, because it should only be returning a `DUPLICATE_FILE` error or a `DUPLICATE_SUPPORT_FILE` error.

Instead, I reversed it so that it returns the `DUPLICATE_SUPPORT_FILE` related error if appropriate, and otherwise the `DUPLICATE_FILE` error by default.

Reply:
Addressing the earlier items -

- added tsdocs on `isDuplicateFileName` and `isDuplicateFolderName`
- Changing the behavior with the red error message is a more involved change, but it's worth looking into. I filed https://codedotorg.atlassian.net/browse/CT-868 as a follow up.